src/djhtmx/query.py

```diff
@@ -138,12 +145,22 @@ def get_updates_for_params(self, value: Any, params: QueryDict) -> list[str]:
         # Otherwise, let's serialize the value and only update it if it is
         # different.
         serialized_value = self.adapter.dump_python(value, mode="json")
+        
+        # For dictionary types the validation on the previous value from the params
+        # fails because it's a string, same as in the update_for_state so, again
+        # I thought I could try parsing it before validating
+        try:
+            from ast import literal_eval
+            assert (val := params.get(self.param_name))
+            param_value = literal_eval(val)
```

Comment:
Let's avoid this.  I would not use eval (not event `literal_eval`) on any user input.